lib/rules/indent.js

```diff
@@ -694,27 +726,27 @@ module.exports = {
                     if (parent.type !== "VariableDeclarator" || parentVarNode === parentVarNode.parent.declarations[0]) {
                         if (parent.type === "VariableDeclarator" && parentVarNode.loc.start.line === parent.loc.start.line) {
                             nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
-                        } else if (
-                            parent.type === "ObjectExpression" ||
-                            parent.type === "ArrayExpression" ||
-                            parent.type === "CallExpression" ||
-                            parent.type === "ArrowFunctionExpression" ||
-                            parent.type === "NewExpression" ||
-                            parent.type === "LogicalExpression"
-                        ) {
-                            nodeIndent = nodeIndent + indentSize;
+                        } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
+                            nodeIndent += options[parent.type] * indentSize;
+                        } else if (parent.type === "CallExpression" || parent.type === "NewExpression") {
+                            nodeIndent += (typeof options.CallExpression.arguments === "number" ? options.CallExpression.arguments : 1) * indentSize;
```

Comment:
Incidentally, is `typeof` checks unnecessary for `ObjectExpression` differently to `CallExpression`?